File: Python/programmers/etc/level4/쿠키_구입.py
def solution(cookie):
    n = len(cookie)
    answer = 0

    for i in range(n-1):
        leftSum, leftIdx = cookie[i], i
        rightSum, rightIdx = cookie[i+1], i+1

        while True:
            if leftSum == rightSum:
                answer = max(answer, leftSum)
            if leftIdx > 0 and leftSum <= rightSum:
                leftIdx -= 1
                leftSum += cookie[leftIdx]
            elif rightIdx < n-1 and rightSum <= leftSum:
                rightIdx += 1
                rightSum += cookie[rightIdx]
            else:
                break

    return answer


cookie = [1,1,2,3]
print(solution(cookie)) # 3

cookie = [1,2,4,5]
print(solution(cookie)) # 0

File: Java/programmers/etc/level4/쿠키구입.java
package CodeTest.Java.programmers.etc.level4;

public class 쿠키구입 {
    public int solution(int[] cookie) {
        int n = cookie.length;
        int answer = 0;

        for (int i=0; i<n-1; i++) {
            int leftSum = cookie[i], leftIdx = i;
            int rightSum = cookie[i+1], rightIdx = i+1;

            while (true) {
                if (leftSum == rightSum) {
                    answer = Math.max(answer, leftSum);
                }
                if (leftIdx > 0 && leftSum <= rightSum) {
                    leftIdx--;
                    leftSum += cookie[leftIdx];
                }
                else if (rightIdx < n-1 && rightSum <= leftSum) {
                    rightIdx++;
                    rightSum += cookie[rightIdx];
                }
                else break;
            }
        }
        
        return answer;
    }
}
